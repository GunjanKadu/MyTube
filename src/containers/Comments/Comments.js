import React from 'react';
import {CommentsHeader} from "./CommentsHeader/CommentsHeader";
import {AddComment} from "./AddComment/AddComment";
import {UserComment} from "./UserComment/UserComment";

export class Comments extends React.Component {
  render() {
    return(
      <div>
        <CommentsHeader amountComments={this.props.amountComments}/>
         <AddComment key='add-comment'/>
        {/* <UserComment/>
        <UserComment/>
        <UserComment/> */}
      </div>
    );
  }
}