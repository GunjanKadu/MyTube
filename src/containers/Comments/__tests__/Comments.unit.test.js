import React from 'react';
import {shallow} from 'enzyme';
import {Comments} from '../Comments';

test('renders Comments with props.comments = null', () => {
  const wrapper = shallow(
    <Comments/>
  );
  expect(wrapper).toMatchSnapshot();
});
