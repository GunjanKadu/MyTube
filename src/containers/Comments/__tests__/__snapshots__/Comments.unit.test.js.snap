// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Comments with props.comments = null 1`] = `<div />`;

exports[`renders Comments with valid props.comments 1`] = `
<React.Fragment>
  <CommentsHeader
    amountComments={3}
  />
  <AddComment
    key="add-comment"
  />
  <UserComment
    comment={
      Object {
        "etag": "\\"XI7nbFXulYBIpL0ayR_gDh3eu1k/7qOYhKtSTLDez0FUuT8Axyuyhvg\\"",
        "id": "UgycuaK9UFzPyBHmgNJ4AaABAg",
        "kind": "youtube#commentThread",
        "snippet": Object {
          "canReply": true,
          "isPublic": true,
          "topLevelComment": Object {
            "etag": "\\"XI7nbFXulYBIpL0ayR_gDh3eu1k/iPL4MvxUFXz1L8Rgn1EAXvIVU5s\\"",
            "id": "UgycuaK9UFzPyBHmgNJ4AaABAg",
            "kind": "youtube#comment",
            "snippet": Object {
              "authorChannelId": Object {
                "value": "UC5hCkka0HvWJ0PqRFhUDaig",
              },
              "authorChannelUrl": "http://www.youtube.com/channel/UC5hCkka0HvWJ0PqRFhUDaig",
              "authorDisplayName": "Peter Unlustig",
              "authorProfileImageUrl": "https://yt3.ggpht.com/-QO3910-PfI0/AAAAAAAAAAI/AAAAAAAAAAA/XiVbzk5N7TI/s28-c-k-no-mo-rj-c0xffffff/photo.jpg",
              "canRate": true,
              "likeCount": 0,
              "publishedAt": "2018-07-27T07:42:12.000Z",
              "textDisplay": "WTF is this setup … what does he need 20 channels for?",
              "textOriginal": "WTF is this setup … what does he need 20 channels for?",
              "updatedAt": "2018-07-27T07:42:12.000Z",
              "videoId": "OVazJLD5lM8",
              "viewerRating": "none",
            },
          },
          "totalReplyCount": 0,
          "videoId": "OVazJLD5lM8",
        },
      }
    }
    key="UgycuaK9UFzPyBHmgNJ4AaABAg"
  />
  <UserComment
    comment={
      Object {
        "etag": "\\"XI7nbFXulYBIpL0ayR_gDh3eu1k/T0sT4IaC2pvS6vcIBKXnPj9Tisc\\"",
        "id": "UgyMcYqdC45ZxlB21od4AaABAg",
        "kind": "youtube#commentThread",
        "snippet": Object {
          "canReply": true,
          "isPublic": true,
          "topLevelComment": Object {
            "etag": "\\"XI7nbFXulYBIpL0ayR_gDh3eu1k/7F2iala6QM8arnhO0vYhucJSNUM\\"",
            "id": "UgyMcYqdC45ZxlB21od4AaABAg",
            "kind": "youtube#comment",
            "snippet": Object {
              "authorChannelId": Object {
                "value": "UCfj2YFpHr-7Cqw_ZGh3iRVg",
              },
              "authorChannelUrl": "http://www.youtube.com/channel/UCfj2YFpHr-7Cqw_ZGh3iRVg",
              "authorDisplayName": "skinnyguy 96",
              "authorProfileImageUrl": "https://yt3.ggpht.com/-3UpLc3OFFg0/AAAAAAAAAAI/AAAAAAAAAAA/ItLZkpJpTIk/s28-c-k-no-mo-rj-c0xffffff/photo.jpg",
              "canRate": true,
              "likeCount": 0,
              "publishedAt": "2018-07-27T03:34:28.000Z",
              "textDisplay": "What&#39;s the song at minute <a href=\\"https://www.youtube.com/watch?v=OVazJLD5lM8&amp;t=33m00s\\">33:00</a> ? Pleaaase",
              "textOriginal": "What's the song at minute 33:00 ? Pleaaase",
              "updatedAt": "2018-07-27T03:34:28.000Z",
              "videoId": "OVazJLD5lM8",
              "viewerRating": "none",
            },
          },
          "totalReplyCount": 0,
          "videoId": "OVazJLD5lM8",
        },
      }
    }
    key="UgyMcYqdC45ZxlB21od4AaABAg"
  />
  <UserComment
    comment={
      Object {
        "etag": "\\"XI7nbFXulYBIpL0ayR_gDh3eu1k/oCeh5wWi-o4DTu_S17Dtg6mwu58\\"",
        "id": "UgyWzeAFVcriz6eoqHJ4AaABAg",
        "kind": "youtube#commentThread",
        "snippet": Object {
          "canReply": true,
          "isPublic": true,
          "topLevelComment": Object {
            "etag": "\\"XI7nbFXulYBIpL0ayR_gDh3eu1k/ry0MKR74RE4suvQiOi4sUze35Ow\\"",
            "id": "UgyWzeAFVcriz6eoqHJ4AaABAg",
            "kind": "youtube#comment",
            "snippet": Object {
              "authorChannelId": Object {
                "value": "UC9chIogLbG5wIL7PVZvu5GQ",
              },
              "authorChannelUrl": "http://www.youtube.com/channel/UC9chIogLbG5wIL7PVZvu5GQ",
              "authorDisplayName": "CyberPixl Music",
              "authorProfileImageUrl": "https://yt3.ggpht.com/-IQnM1McHOtU/AAAAAAAAAAI/AAAAAAAAAAA/bg0izTzgGVw/s28-c-k-no-mo-rj-c0xffffff/photo.jpg",
              "canRate": true,
              "likeCount": 1,
              "publishedAt": "2018-07-27T00:35:42.000Z",
              "textDisplay": "<b><i>You just gotta love how he feels the music!</i></b>",
              "textOriginal": "*_You just gotta love how he feels the music!_*",
              "updatedAt": "2018-07-27T00:35:42.000Z",
              "videoId": "OVazJLD5lM8",
              "viewerRating": "none",
            },
          },
          "totalReplyCount": 0,
          "videoId": "OVazJLD5lM8",
        },
      }
    }
    key="UgyWzeAFVcriz6eoqHJ4AaABAg"
  />
</React.Fragment>
`;
